deps/chakrashim/src/jsrtisolateshim.cc

```diff
@@ -670,6 +679,40 @@ void IsolateShim::SetPromiseRejectCallback(v8::PromiseRejectCallback callback) {
                                    reinterpret_cast<void*>(callback));
 }
 
+void IsolateShim::RunMicrotasks() {
+  // Loop until we've handled all the tasks, including the ones added by these tasks
+  while(this->microtaskQueue != nullptr) {
+    MicroTask* batch = this->microtaskQueue;
+    this->microtaskQueue = nullptr;
+
+    // Ensure that we handle things in-order
+    MicroTask* ordered = nullptr;
```

Comment:
I don't follow- what is the intent here?

Reply:
Can we use a tail pointer and append to the end (i.e implement an actual queue)

I'd also just settle for a less terse comment

Reply:
Updated the comment